MinersLegacy/SRBMiner.ps1

```diff
@@ -0,0 +1,86 @@
+    [PSCustomObject]@{Algorithm = "marketcash"; Threads = 1; Params = @()} # CryptoNight-MarketCash 1 thread
+    [PSCustomObject]@{Algorithm = "normalv7"  ; Threads = 1; Params = @()} # CryptoNightV7 1 thread
+    [PSCustomObject]@{Algorithm = "stellitev4"; Threads = 1; Params = @()} # CryptoNight-Stellite 1 thread
+    [PSCustomObject]@{Algorithm = "alloy"     ; Threads = 2; Params = @()} # CryptoNight-Alloy 2 threads
+    [PSCustomObject]@{Algorithm = "artocash"  ; Threads = 2; Params = @()} # CryptoNight-ArtoCash 2 threads
+    [PSCustomObject]@{Algorithm = "b2n"       ; Threads = 2; Params = @()} # CryptoNight-B2N 2 threads
+    [PSCustomObject]@{Algorithm = "fast"      ; Threads = 2; Params = @()} # CryptoNight-Fast (Masari) 2 threads
+    [PSCustomObject]@{Algorithm = "lite"      ; Threads = 2; Params = @()} # CryptoNight-Lite 2 threads
+    [PSCustomObject]@{Algorithm = "heavy"     ; Threads = 2; Params = @()} # CryptoNight-Heavy 2 threads
+    [PSCustomObject]@{Algorithm = "haven"     ; Threads = 2; Params = @()} # CryptoNight-Haven 2 threads
+    [PSCustomObject]@{Algorithm = "ipbc"      ; Threads = 2; Params = @()} # CryptoNight-PPBC 2 threads
+    [PSCustomObject]@{Algorithm = "marketcash"; Threads = 2; Params = @()} # CryptoNight-MarketCash 2 threads
+    [PSCustomObject]@{Algorithm = "normalv7"  ; Threads = 2; Params = @()} # CryptoNightV7 2 thread
+    [PSCustomObject]@{Algorithm = "stellitev4"; Threads = 1; Params = @()} # CryptoNight-Stellite 2 threads
+    #[PSCustomObject]@{Algorithm = "normal"    ; Threads = 1; Params = @()} # CryptoNight 1 thread, ASIC territory
+)
+
+$Name = Get-Item $MyInvocation.MyCommand.Path | Select-Object -ExpandProperty BaseName
+$Devices = $Devices | Where-Object Type -EQ "GPU" | Where-Object Vendor -EQ "Advanced Micro Devices, Inc."
+
+$Devices | Select-Object Vendor, Model -Unique | ForEach-Object {
+    $Miner_Device = $Devices | Where-Object Model -EQ $_.Model
+    $Miner_Port = $Port -f ($Miner_Device | Select-Object -First 1 -ExpandProperty Index)
+
+    $Commands | ForEach-Object {
+        $Algorithm = $_.Algorithm
+        $Algorithm_Norm = Get-Algorithm "cryptonight-$($Algorithm)"
+        $Threads = $_.Threads
+        $Params = $_.Params
+        $Miner_Name = (@($Name) + @($Threads) + @($Miner_Device.Model_Norm | Sort-Object -Unique) | Select-Object -Unique) -join '-'
+
+        $HashRate = $Stats."$Miner_Name".Week
+
+        $ConfigFile = "$($Miner_Name)-$($Algorithm)-Port$($Miner_Port).json"
+      
+        (
+            [PSCustomObject]@{
+                api_enabled      = $true
+                api_port         = [Int]$Miner_Port
+                api_rig_name     = "$($Config.Pools.$($Pools.$Algorithm_Norm.Name).Worker)"
+                cryptonight_type = $Algorithm
+                gpu_conf         = @([PSCustomObject]@{
+                    "id"        = [Int]$Miner_Device.PlatformId_Index  
```

Comment:
@Aaron: Is this now the correct index?

Reply:
Yes.

Reply:
No, sorry, this is completely wrong. You are converting an 'array' to an 'int' plus this miner does not appear to use platforms (or at least you haven't set one).

Reply:
very true... changed.